src/NinjaTools.FluentMockServer/FluentAPI/Builders/HttpEntities/FluentHttpRequestBuilder.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using JetBrains.Annotations;
+using Newtonsoft.Json.Linq;
+using NinjaTools.FluentMockServer.FluentAPI.Builders.ValueObjects;
+using NinjaTools.FluentMockServer.Models.HttpEntities;
+using NinjaTools.FluentMockServer.Models.ValueTypes;
+
+namespace NinjaTools.FluentMockServer.FluentAPI.Builders.HttpEntities
+{
+    internal sealed class FluentHttpRequestBuilder : IFluentHttpRequestBuilder
+    {
+        [CanBeNull] private string _method;
+        [CanBeNull] private Dictionary<string, string[]> _headers;
+        [CanBeNull] private JToken _body;
+        [CanBeNull] private string _path;
+        private bool? _secure;
+        private bool? _keepAlive;
+        
+        // TODO: Add Builder Methods
+        // ReSharper disable once UnusedAutoPropertyAccessor.Local
+        [CanBeNull] private Dictionary<string, string> Cookies { get; set; }
+
+        /// <inheritdoc />
+        public IFluentHttpRequestBuilder WithMethod(HttpMethod method)
+        {
+            return WithMethod(method.Method);
+        }
+
+        /// <inheritdoc />
+        [NotNull]
+        public IFluentHttpRequestBuilder WithMethod(string method)
+        {
+            _method = method;
+            return this;
+        }
+
+        /// <inheritdoc />
+        [NotNull]
+        public IFluentHttpRequestBuilder WithPath([CanBeNull] string path)
+        {
+            _path = path?.TrimStart('/').Insert(0, "/") ?? "/";
+            return this;
+        }
+
+        /// <inheritdoc />
+        [NotNull]
+        public IFluentHttpRequestBuilder KeepConnectionAlive(bool keepalive = true)
+        {
+            _keepAlive = keepalive;
+            return this;
+        }
+
+        /// <inheritdoc />
+        [NotNull]
+        public IFluentHttpRequestBuilder EnableEncryption(bool useSsl = true)
+        {
+            _secure = true;
+            return this;
+        }
+
+        /// <inheritdoc />
+        [NotNull]
+        public IFluentHttpRequestBuilder WithBody(Body.BodyType type, string value)
+        {
+            var builder = new FluentBodyBuilder();
+
+            switch (type)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)